lib/Driver/ToolChains.h

```diff
@@ -50,6 +50,20 @@ class LLVM_LIBRARY_VISIBILITY GenericUnix : public ToolChain {
   ~GenericUnix() = default;
 };
 
+class LLVM_LIBRARY_VISIBILITY Android : public ToolChain {
```

Comment:
I think it would make more sense to inherit from GenericUnix. It seems fair to even make an overload point for LinkJobAction if the Android customization is easy to subset out.